internal/transport/controlbuf.go

```diff
@@ -990,8 +975,31 @@ func (l *loopyWriter) processData() (bool, error) {
 	if dataItem.onEachWrite != nil {
 		dataItem.onEachWrite()
 	}
-	if err := l.framer.fr.WriteData(dataItem.streamID, endStream, buf[:size]); err != nil {
-		return false, err
+
+	switch fr := l.framer.fr.(type) {
+	case *Framer:
```

Comment:
here's an idea, how about we make the file `internal/transport/framer.go` part of a package called `grpcframer`. That way this line would look like `grpcframer.Framer` which makes it more readable.